docs/source/using_yosys/more_scripting/data_flow_tracking.rst

```diff
@@ -0,0 +1,71 @@
+For example, the state of a bit could be ``0`` and the set of tags ``"KEY"`` and ``"OVERFLOW"``.
+
+In addition to their usual operations on the logical bits, Yosys operations must now also process the status of the tags.
+For this, tags are simply *forwarded* or *propagated* (i.e. copied) from inputs to outputs, according to the following general rule:
+
+   A tag is forwarded from an input to an output if the input can affect the output, for that particular state of all other inputs.
+
+For example, XOR, AND and OR cells propagate tags as follows:
+
+#. XOR simply forwards all tags from its inputs to its output, because inputs to XOR can always affect the output.
+#. AND forwards tags only if the other input is ``1``. Because if one input is ``0``, the other input can never affect the output.
+#. Similarly, OR forwards tags only if the other input is ``0``.
+
+There are two exceptions to this rule:
+#. In general, propagation is only determined approximately.
+   For example, unless the ``dft_tag`` code knows about a cell, it simply assumes the worst-case behaviour that all inputs can affect all outputs.
+   It also does not try to determine whether the effect of a signal is cancelled out, for example ``A ^ A`` or ``A ^ (B ^ A)`` is independent of ``A``, but its tags would be propagated nonetheless.
+#. If tag groups are used, the rules are modified (see below).
+
+Because of this propagation behaviour, we can answer questions about what signals are affected by a certain signal, by injecting a tag at that point in the circuit, and observing where the tag is visible.
+
+Example use cases
+~~~~~~~~~~~~~~~~~~
+As an example use case, consider a cryptographic processor which is not supposed to expose its secret keys to the outside world.
+We can tag all key bits with the ``"KEY"`` tag and use ``sby`` to formally verify that no external signal ever carries the ``"KEY"`` tag, meaning that key information is not visible to the outside.
+As a caveat, we have to manually clear the ``"KEY"`` tag during cryptographic operations, as proving that the cryptographic operations themselves do not leak key information is beyond the ability of Yosys.
+However we can still easily detect, if e.g. an engineer forgot to remove debugging code that allows reading back key data.
+
+As a different use case, we can modify all adders in the design to set the ``"OVERFLOW"`` tag on their output bits, if the addition overflowed, and then add asserts to all flip-flop inputs and output signals that they do not carry the ``"OVERFLOW"`` tag, i.e. that the results of overflowed additions never affect system state.
+Note that in this particular example we use the ability of tag insertion to be conditional on logic, in this case the overflow condition of an adder.
+
+Semantics of dataflow tracking cells
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``$set_tag`` has inputs ``A``, ``SET``, ``CLR``, an output ``Y`` and a string parameter ``TAG``.
```

Comment:
This is the only heading with an empty line after it, the rest just go straight to content. For consistency with the rest of the documentation I would suggest adding empty lines after all the headings.